enterprise/backend/src/metabase_enterprise/semantic_search/env.clj

```diff
@@ -1,17 +1,18 @@
 (ns metabase-enterprise.semantic-search.env
   (:require
-   [metabase-enterprise.semantic-search.db :as semantic.db]
+   [metabase-enterprise.semantic-search.db.datasource :as semantic.db.datasource]
    [metabase-enterprise.semantic-search.embedding :as semantic.embedding]
    [metabase-enterprise.semantic-search.index-metadata :as semantic.index-metadata]))
 
 ;; global configuration / state containment facility
 ;; only core and the task definitions (for which integration with metabase demands globals)
 ;; should need these
 
+;; TODO: Probably remove completely, all occurrences should be using db.connection
```

Comment:
I would prefer connection functions, conn pool configuration functions _also_ were not about any particular global variable and basically _nothing_ other than the core hooks / task defs with metabase core needs particular globals. 

This is a debate we can have another time, just an opinion from me - others may prefer the globals. Up to you whether you want to keep the comment for now!